lib/listener.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const co = require('co');
+const EventEmitter = require('events');
+const bus = require('./client');
+
+/**
+ * Return a bus listener with helper methods to register listeners and listen to the bus' messages
+ * The instance inherits EventEmitter and can emit following events :
+ * - connected : when the listener is connected to the bus, takes no arguments
+ *
+ * @return {Object} bus listener instance
+ */
+function* createBusListener(url) {
+  const _queues = [];
+  const _handlers = {};
```

Comment:
instance is exported, queues and handlers are implementation details, so they do not need to be put into instance